### ISSUE TITLE
Material-UI's Drawer component causes props are not supported: `ref` error on first open.

### ISSUE THREAD

**Issue by joepagan** [preactjs/preact#2682]
Hi, I hope this is enough information, let me know otherwise. Thanks for your work.

This looks like another issue had the identical error in it at one point:
https://github.com/preactjs/preact/issues/2249 

Another with a similar error https://github.com/preactjs/preact/issues/1824

### Reproduction

I tried to set this up in a codesandbox app but kept getting "VNode is undefined" errors with MUI components, so I've created a repo for you instead:
https://github.com/joepagan/preact-mui-ref-issue

### Steps to reproduce

1. `npm run dev` & load http://0.0.0.0:8080
2. click menu icon in top-right
3. this should open the [temporary Drawer component](https://material-ui.com/components/drawers/#temporary-drawer)
4. Drawer opens with console error

### Expected Behavior

no console error

### Actual Behavior

console error on draw open, it does not occur again when closed or re-opened:

```
Failed prop type: The following props are not supported: `ref`. Please remove them. check-props.js:43
    __b check-props.js:43
    __b check-props.js:25
    __b forwardRef.js:6
    T index.js:70
    g children.js:143
    T index.js:187
    g children.js:143
    T index.js:187
    g children.js:143
    T index.js:187
    g children.js:143
    T index.js:187
    g children.js:143
    j index.js:390
    T index.js:237
    g children.js:143
    j index.js:390
    T index.js:237
    g children.js:143
    T index.js:187
    g children.js:143
    T index.js:187
    g children.js:143
    j index.js:390
    T index.js:237
    g children.js:143
    T index.js:187
    g children.js:143
    T index.js:187
    g children.js:143
    T index.js:187
    g children.js:143
    T index.js:187
    g children.js:143
    j index.js:390
    T index.js:237
    g children.js:143
    T index.js:187
    m component.js:216
    m component.js:211
    (Async: promise callback)
    w component.js:194
    setState component.js:30
    setState debug.js:340
    __ index.js:134
    toggleDrawer index.js:61
    N props.js:159
    (Async: EventListener.handleEvent)
    C props.js:90
    A props.js:24
    j index.js:390
    T index.js:237
    g children.js:143
    T index.js:187
    g children.js:143
    T index.js:187
    g children.js:143
    T index.js:187
    g children.js:143
    T index.js:187
    g children.js:143
    T index.js:187
    g children.js:143
    j index.js:390
    T index.js:237
    g children.js:143
    j index.js:390
    T index.js:237
    g children.js:143
    T index.js:187
    g children.js:143
    T index.js:187
    g children.js:143
    j index.js:390
    T index.js:237
    g children.js:143
    T index.js:187
    g children.js:143
    T index.js:187
    g children.js:143
    T index.js:187
    g children.js:143
    T index.js:187
    g children.js:143
    j index.js:390
    T index.js:237
    g children.js:143
    T index.js:187
    g children.js:143
    j index.js:390
    T index.js:237
    g children.js:143
    T index.js:187
    g children.js:143
    T index.js:187
    M render.js:17
    Preact 2
    js bundle.js:3439
    __webpack_require__ bundle.js:833
    fn bundle.js:130
    1 bundle.js:4259
    __webpack_require__ bundle.js:833
    <anonymous> bundle.js:971
    <anonymous> bundle.js:974
```

**Comment by sventschui** [comment]
Hi @joepagan 

Thanks for your bug report and the repro!

This warning comes from `preact/debug`. Despite the message being logged certainly is a bug, it won't affect the functionality of your application, or does it?

I'll investigate why we print this warning and whether we can suppress it in this case

**Comment by sventschui** [comment]
With #2685 (& `@babel/plugin-transform-react-jsx-source`) the error looks like this:

```
check-props.js?860e:43 Failed prop type: The following props are not supported: `ref`. Please remove them.
  in ForwardRef(Portal)
  in ForwardRef(Modal)
  in ForwardRef(Drawer)
  in WithStyles(ForwardRef(Drawer)) (at .../preact-mui-ref-issue/src/components/header/index.js:127)
  in ButtonAppBar (at .../preact-mui-ref-issue/src/components/app.js:22)
  in App```

Something is going wrong in prop type validation of a forwardRef'ed Portal

**Comment by sventschui** [comment]
Pushed a fix to #2685 to no longer pass `ref` property to prop types validation for `forwardRef`ed Components as this breaks material-uis exact prop types validation (and also `airbnb/prop-types-exact`).

**Comment by joepagan** [comment]
Hi @sventschui 

Thanks for getting back, your detailed response about the issue and ofcourse your work on a fix!

> it won't affect the functionality of your application, or does it?

No it doesn't appear to cause an issue.
